agent/lib/kubernetes.js

```diff
@@ -92,12 +92,9 @@ function list_options(resource, baseOptions) {
 
 function watch_options(resource, baseOptions) {
     var base = resource.startsWith("address") ? '/apis/enmasse.io/v1beta1/watch/namespaces/' : '/api/v1/watch/namespaces/';
-    var options = get_options(baseOptions, get_path(base, resource, includeResyncInterval(baseOptions)));
-    return includeResyncInterval(options);
-}
-
-function includeResyncInterval(options) {
-    return myutils.merge({requestTimeout: options.requestTimeout || process.env.RESYNC_INTERVAL}, options);
+    baseOptions = myutils.merge({requestTimeout: baseOptions.requestTimeout || process.env.RESYNC_INTERVAL}, baseOptions);
+    let path = get_path(base, resource, baseOptions);
+    return get_options(baseOptions, path);;
```

Comment:
get_options(..) will remove the requestTimeout field from the returned options.  Having a look to see if it is actually needed on the watch call.

Reply:
Seems like the options that watch_options returns, needs to contain the requestTimeout.